src/Symfony/Component/Cache/Tests/Adapter/MemcachedAdapterTest.php

```diff
@@ -191,4 +192,29 @@ public function provideDsnWithOptions()
             array(\Memcached::OPT_SOCKET_RECV_SIZE => 1, \Memcached::OPT_SOCKET_SEND_SIZE => 2, \Memcached::OPT_RETRY_TIMEOUT => 8),
         );
     }
+
+    public function testNonBinaryMemcachedMode()
```

Comment:
Instead of adding a few new tests for that, covering only a small part of the API, I think we should instead have a separate test case overriding `createCachePool`, to run the full adapter tests in non-binary mode

Reply:
MemcachedAdapterTest tests MemcachedAdapter::createConnection only. Should I add separate test classes to test MemcachedAdapter get, set, etc.. methods in text and binary modes?

Reply:
It looks like I should add key containing \0 and spaces into \Cache\IntegrationTests\SimpleCacheTest::validKeys

This way it would be tested against all cache implementations.

Reply:
Or rely on the namespace to do so?

Reply:
I added 'si\0mp le' key to \Cache\IntegrationTests\SimpleCacheTest::validKeys and ran tests local. All cache tests succeeded.